utilities/api/index.spec.ts

```diff
@@ -1,39 +1,39 @@
 import { CONTENTFUL_API_URL } from './constants'
-import { getAllPageContentsWithSlugString, getPageContentBySlugString } from './index'
+import { getAllPathnames, getPageContentByPathname } from './'
 import type { PageContentProps } from '../../components/Main/PageContent'
-import type { PageContentsWithSlugString } from '../../pages/[[...slug]]'
+import type { Pathname } from './'
 import { rest } from 'msw'
 import { setupServer } from 'msw/node'
 import 'whatwg-fetch'
 
-const pageContent: PageContentProps = {
-  content: 'mock content',
-  title: 'mock title',
-}
-const allPageContentsWithSlug: PageContentsWithSlugString[] = [
+const mockAllPathnames: Pathname[] = [
   {
-    slugString: 'demo/js',
+    pathname: 'demo/js',
   },
   {
-    slugString: 'demo/python',
+    pathname: 'demo/python',
   },
 ]
+const mockPageContent: PageContentProps = {
+  content: 'mock content',
+  title: 'mock title',
+}
 
+// todo: use mock graphql
 const server = setupServer(
   rest.post(CONTENTFUL_API_URL, (req, res, ctx) => {
-    const query = (req?.body as Record<string, string>)?.query
-    let items
+    const query = (req.body as Record<string, string>).query
+    let items: PageContentProps[] | Pathname[]
```

Comment:
Still have one `let` in the test. This part should be rewritten using mocking GraphQL API from MSW. So this `let` will be removed then. 
But still curious, how to avoid using let in this scenario：
```js
let x
switch (someString) {
  case 'apple':
    x = 1
    break
  case 'banana':
    x = 0
    break
  case 'orange':
    x = 0
    break
}
```

Reply:
Move the switch into a function:

```typescript
function setXByFruit(fruit) {
  switch (fruit) {
    case 'apple':
      return 1
    case 'banana':
    case 'orange':
      return 0
  }
}
```

Then wherever you need to set x:

```typescript
  const x = setXByFruit(fruit)
```

Reply:
I'm looking at your actual code now and this is a very bad use of `switch`. If your switch is on `true`, then aren't you just testing conditions? The proper syntax for that is if/else. You can still move it into a function (you'll need to).

A general rule: if you're doing something very clever, then you're doing something very wrong. Good code is not clever, it's obvious and easy to understand. Despite having programmed for forty years in a couple dozen different languages, I had to think about the code for a moment to figure out what the hell it was doing. That's not a code smell. It's a code _stink_.

I know clever code looks fun, but trust me: [Murder _all_ your darlings](https://tinhouse.com/murder-your-darlings/).